### PR TITLE
Add JsonReader.skipName.

### LINKED ISSUE #518: Add JsonReader.skipName().

### PR DESCRIPTION
**Pull request by NightlyNexus**
Closes #518.

### CODE REVIEW

File: moshi/src/test/java/com/squareup/moshi/MoshiTest.java [hunk 1]
```diff
@@ -910,7 +910,7 @@
       adapter.fromJson("{\"diameter\":5,\"crust\":\"thick\",\"extraCheese\":true}");
       fail();
     } catch (JsonDataException expected) {
-      assertThat(expected).hasMessage("Cannot skip unexpected STRING at $.crust");
+      assertThat(expected).hasMessage("Cannot skip unexpected NAME at $.diameter");
```
Comment (NightlyNexus) [inline moshi/src/test/java/com/squareup/moshi/MoshiTest.java]: hm, not sure if there's something to do better here.

File: kotlin/src/main/java/com/squareup/moshi/kotlin/KotlinJsonAdapter.kt [hunk 2]
```diff
@@ -68,7 +68,7 @@ internal class KotlinJsonAdapter<T>(
       val binding = if (index != -1) bindings[index] else null
 
       if (binding == null) {
-        reader.nextName()
+        reader.skipName()
```
Comment (swankjesse) [inline kotlin/src/main/java/com/squareup/moshi/kotlin/KotlinJsonAdapter.kt]: nice

File: moshi/src/main/java/com/squareup/moshi/JsonUtf8Reader.java [hunk 3]
```diff
@@ -562,6 +562,27 @@ private boolean isLiteral(int c) throws IOException {
...
+      skipUnquotedValue();
+    } else if (p == PEEKED_DOUBLE_QUOTED_NAME) {
+      skipQuotedValue(DOUBLE_QUOTE_OR_SLASH);
+    } else if (p == PEEKED_SINGLE_QUOTED_NAME) {
+      skipQuotedValue(SINGLE_QUOTE_OR_SLASH);
+    } else if (p != PEEKED_BUFFERED_NAME) {
+      throw new JsonDataException("Expected a name but was " + peek() + " at path " + getPath());
+    }
+    peeked = PEEKED_NONE;
+    pathNames[stackSize - 1] = "null";
```
Comment (swankjesse) [inline moshi/src/main/java/com/squareup/moshi/JsonUtf8Reader.java]: this is interesting!

Comment (NightlyNexus) [reply moshi/src/main/java/com/squareup/moshi/JsonUtf8Reader.java]: seemed better than leaving it empty.

**Pull request merged**